### PR TITLE
Refactor alliance benchmarks to v2

### PR DESCRIPTION
**Pull request by 0xmovses**
- This PR refactors `alliance/src/benchmarkings.rs` to use benchmarking v2. These changes are needed to improve the readability and maintainability of the benchmarking code. 

- No known issue to backlink.

## Local Testing 
1. `cargo build --features runtime-benchmarks` 
2. `cargo run --locked --release -p node-cli --bin substrate-node --features runtime-benchmarks -- benchmark pallet --execution wasm --wasm-execution compiled --chain dev --pallet "*" --extrinsic "*" --steps 2 --repeat 1`

### CODE REVIEW

**Comment by 0xmovses** [comment]
bot fmt

**Review by sam0x17 — APPROVED** [review]

**Review by seadanda — APPROVED** [review]

File: substrate/frame/alliance/src/benchmarking.rs [hunk 1]
```diff
@@ -726,58 +765,71 @@ benchmarks_instance_pallet! {
...
-			member: fellow2,
-			slashed: Some(T::AllyDeposit::get())
-		}.into());
+		assert_last_event::<T, I>(
+			Event::MemberKicked { member: fellow2, slashed: Some(T::AllyDeposit::get()) }.into(),
+		);
+		Ok(())
 	}
 
-	add_unscrupulous_items {
```
Comment (ggwpez) [inline substrate/frame/alliance/src/benchmarking.rs:741]: It looks like `add_unscrupulous_items` was forgotten. Could you please re-add it @0xmovses?

Comment (0xmovses) [reply substrate/frame/alliance/src/benchmarking.rs:741]: Ah, yes sure thing.

Comment (0xmovses) [reply substrate/frame/alliance/src/benchmarking.rs:741]: @ggwpez It's there, see here on master https://github.com/paritytech/polkadot-sdk/blob/7b06e634fe749508b639e50cccbeb58ce794323c/substrate/frame/alliance/src/benchmarking.rs#L785

Comment (ggwpez) [reply substrate/frame/alliance/src/benchmarking.rs:741]: `add_unscrupulous_items` != `add_scrupulous_items`

Comment (0xmovses) [reply substrate/frame/alliance/src/benchmarking.rs:741]: adding it back now.

Comment (0xmovses) [reply substrate/frame/alliance/src/benchmarking.rs:741]: @ggwpez so it seems that there was never an `add_scrupulous_items`, but only a `add_unscruplous_items`, see master branch before this PR was merged https://github.com/paritytech/polkadot-sdk/blob/cfb29254f74412cea35e8048d8aea94bc789fcb1/substrate/frame/alliance/src/benchmarking.rs#L741
 in this PR, it seems the name of the function was wrongly changed. 
  
 Put up a fix here https://github.com/paritytech/polkadot-sdk/pull/2173

**Pull request merged**